novawallet/Common/Services/RemoteSubscription/Substrate/Multisig/PendingOperationChainSync/PendingMultisigLocalStorageSyncService.swift

```diff
@@ -0,0 +1,431 @@
+        remotePendingOperations: MultisigPendingOperationsMap
+    ) -> CompoundOperationWrapper<Set<CallHash>> {
+        let localFetchOperation = pendingOperationsRepository.fetchAllOperation(with: .init())
+
+        let diffOperation = ClosureOperation<CallHashChanges> {
+            let oldCallKeys = try localFetchOperation
+                .extractNoCancellableResultData()
+                .map { $0.createKey() }
+
+            let updatedKeys = remotePendingOperations.map(\.key)
+
+            let updatedKeySet = Set(updatedKeys)
+            let oldKeySet = Set(oldCallKeys)
+
+            return CallHashChanges(
+                allOperationsKeys: updatedKeySet,
+                removedOperationsKeys: oldKeySet.subtracting(updatedKeySet)
+            )
+        }
+
+        let updateOperation = createUpdateOperation(
+            remotePendingOperations: remotePendingOperations,
+            localFetchClosure: { try localFetchOperation.extractNoCancellableResultData() },
+            removedOperationsClosure: {
+                try diffOperation
+                    .extractNoCancellableResultData()
+                    .removedOperationsKeys
+            }
+        )
+
+        let resultOperation = ClosureOperation<Set<CallHash>> {
+            try updateOperation.extractNoCancellableResultData()
+
+            let callHashes = try diffOperation
+                .extractNoCancellableResultData()
+                .allOperationsKeys
+                .map(\.callHash)
+
+            return Set(callHashes)
+        }
+
+        diffOperation.addDependency(localFetchOperation)
+        updateOperation.addDependency(diffOperation)
+        resultOperation.addDependency(updateOperation)
+
+        return CompoundOperationWrapper(
+            targetOperation: resultOperation,
+            dependencies: [localFetchOperation, diffOperation, updateOperation]
+        )
+    }
+
+    func createUpdateOperation(
+        remotePendingOperations: MultisigPendingOperationsMap,
+        localFetchClosure: @escaping () throws -> [Multisig.PendingOperation],
+        removedOperationsClosure: @escaping () throws -> Set<Multisig.PendingOperation.Key>
+    ) -> BaseOperation<Void> {
+        pendingOperationsRepository.saveOperation(
+            {
+                let localPendingOperations = try localFetchClosure()
+                    .reduce(into: [:]) { $0[$1.createKey()] = $1 }
+
+                let newItems = remotePendingOperations.filter { localPendingOperations[$0.key] == nil }.values
+
+                let updates: [Multisig.PendingOperation] = localPendingOperations.compactMap { key, local in
+                    guard let remote = remotePendingOperations[key] else { return nil }
+
+                    let definitionUpdated = remote.multisigDefinition != local.multisigDefinition
+                    let callUpdated = remote.call != nil && local.call == nil
+
+                    guard definitionUpdated || callUpdated else { return nil }
```

Comment:
I didn't get why do we tweak the models to save as we do the diff in the previous step?